src/types.js

```diff
@@ -2,8 +2,41 @@
 
 import { type Schema } from "normalizr";
 
-export type SchemaMapType = {
-  [schemaName: string]: Schema
+export type SchemaMapType<SchemasType: string> = {
+  [schemaName: SchemasType]: Schema
```

Comment:
No, it will always be a string, but it will never be **any string**
All types and functions are now polymorphic in respect to the schemas they are scoped with
This greatly improves the typing since flow can understand that only the keys of the passed `schemaMap` flow through the code as `schema: SchemasType`
See https://flow.org/blog/2015/03/12/Bounded-Polymorphism/